### ISSUE TITLE
multi-level model error

### ISSUE THREAD

**Issue by jrosen48** [data-edu/data-science-in-education#302]
I am flummoxed by a multi-level model error. 

When running the analysis with all of the data, an error is returned (`m1`). 

With different subsets, the model runs fine (`m2` and `m3`). 

With the values between the two subsets that work removed, an error is returned again (`m4`).

I looked at the data - and the subset of the data associated with the problem - and don't see anything atypical.

``` r
library(lme4)
#> Loading required package: Matrix
#devtools::install_github("data-edu/dataedu")
library(dataedu)

dat <- dataedu::sci_mo_processed

m1 <- lmer(percentage_earned ~ 1 + (1|course_id), data = dat)
#> boundary (singular) fit: see ?isSingular
m2 <- lmer(percentage_earned ~ 1 + (1|course_id), data = dat[1:243, ])
m3 <- lmer(percentage_earned ~ 1 + (1|course_id), data = dat[287:603, ])
m4 <- lmer(percentage_earned ~ 1 + (1|course_id), data = dat[-c(244:286), ])
#> boundary (singular) fit: see ?isSingular
```

<sup>Created on 2020-02-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by jrosen48** [comment]
A few other things I tried:

- calculating the mean per group (nothing stood out, apart from one group with only one case, and two groups with two cases; removing those didn't solve the problem)
- plotting the distribution of the DV values per group

**Comment by datalorax** [comment]
Hi Josh,

I think the issue here is that you're trying to model a proportion as the dependent variable in a linear model. See [here](https://stats.stackexchange.com/questions/103731/what-are-the-issues-with-using-percentage-outcome-in-linear-regression) for why that is problematic, and [here](https://stats.stackexchange.com/questions/87956/how-to-apply-binomial-glmm-glmer-to-percentages-rather-than-yes-no-counts) for different ways of approaching this with `glmer` instead of `lmer`.  Sometimes you can get around this by just transforming the variable but that's usually not ideal.

I can try to circle back later to see if I can get the warnings to go away, but hopefully those resources are helpful in the interim.

**Comment by datalorax** [comment]
Alright, sorry, I didn't look at this close enough. I think the second link I have above is directly applicable here. So your m1 would become

```r
m1 <- glmer(total_points_earned / total_points_possible ~ 1 + (1|course_id), 
            weights = total_points_possible,
            family = binomial,
            data = dat)
```

And that estimates without issue. Should work similarly for the others (although I'm not sure I understand why you're taking out rows?).

**Comment by jrosen48** [comment]
thank you so, so much @datalorax for both helping me to see why this is an issue and a way forward. keeping this open as I want to integrate this into the book. thanks again

**Comment by datalorax** [comment]
No problem! Glad it was helpful.

**Comment by jrosen48** [comment]
cc/ #163

**Comment by restrellado** [comment]
Addressed in #303
